src/Draco.Compiler/Internal/Documentation/DocumentationElement.cs

```diff
@@ -0,0 +1,120 @@
+using System.Collections.Immutable;
+using System.Linq;
+using System.Reflection.Metadata;
+using System.Xml.Linq;
+using Draco.Compiler.Internal.Symbols;
+using Draco.Compiler.Internal.Symbols.Metadata;
+
+namespace Draco.Compiler.Internal.Documentation;
+
+/// <summary>
+/// Represents single documentation element.
+/// </summary>
+internal abstract record class DocumentationElement
+{
+    /// <summary>
+    /// Creates a markdown representation of this documentation element.
+    /// </summary>
+    /// <returns>The documentation in markdown format.</returns>
+    public abstract string ToMarkdown();
+
+    /// <summary>
+    /// Creates an XML representation of this documentation element.
+    /// </summary>
+    /// <returns>The documentation in XML format.</returns>
+    public abstract XNode ToXml();
+}
+
+/// <summary>
+/// Represents regular text inside documentation.
+/// </summary>
+/// <param name="Text">The text represented by this element.</param>
+internal sealed record class TextDocumentationElement(string Text) : DocumentationElement
+{
+    public override string ToMarkdown() => this.Text;
+
+    public override XText ToXml() => new XText(this.Text);
+}
+
+/// <summary>
+/// A single parameter.
+/// </summary>
+/// <param name="ParameterLink">The link to given parameter.</param>
+/// <param name="Elements">The <see cref="DocumentationElement"/>s that are contained in the description of this parameter.</param>
+internal sealed record class ParameterDocumentationElement(ParameterSymbol? Parameter, ImmutableArray<DocumentationElement> Elements) : DocumentationElement
```

Comment:
If you introduce a common, abstract base class like `SymbolDocumentationElement`, you could get rid of all this repetition